src/Catalyst.Core.Lib.Tests/UnitTests/Validators/TransactionValidatorTests.cs

```diff
@@ -101,64 +85,93 @@ public void
             {
-                PublicEntries =
+                Base = new BaseEntry
                 {
-                    new PublicEntry
-                    {
-                        Base = new BaseEntry
-                        {
-                            SenderPublicKey = privateKey.GetPublicKey().Bytes.ToByteString()
-                        }
-                    }
+                    SenderPublicKey = privateKey.GetPublicKey().Bytes.ToByteString()
                 }
             };
 
             var txSig = new Signature
             {
-                RawBytes = new byte[64].ToByteString(), //random bytes that are not of a signed TransactionBroadcast Object
+                RawBytes = new byte[64]
+                   .ToByteString(), //random bytes that are not of a signed TransactionBroadcast Object
                 SigningContext = new SigningContext
                 {
                     NetworkType = NetworkType.Devnet,
                     SignatureType = SignatureType.TransactionPublic
                 }
             };
-            
-            subbedContext.GetSignatureFromBytes(Arg.Is(txSig.ToByteArray()), Arg.Is(privateKey.GetPublicKey().Bytes))
-               .ReturnsForAnyArgs(Substitute.For<ISignature>());
-            
-            subbedContext.Verify(Arg.Any<ISignature>(), // @TODO be more specific
-                    Arg.Is(validTransactionBroadcast.ToByteArray()),
-                    Arg.Is(txSig.SigningContext.ToByteArray())
-                )
-               .Returns(false);
-            
-            subbedContext.Sign(Arg.Is(privateKey), validTransactionBroadcast.ToByteArray(), txSig.SigningContext.ToByteArray())
-               .SignatureBytes.Returns(new byte[64]);
-            
+
             var transactionValidator = new TransactionValidator(subbedLogger, subbedContext);
 
-            validTransactionBroadcast.Signature = txSig;
+            validTransaction.Signature = txSig;
 
-            var result = transactionValidator.ValidateTransaction(validTransactionBroadcast);
+            var result = transactionValidator.ValidateTransaction(validTransaction);
             result.Should().BeFalse();
         }
+
+        private sealed class FakeContext : ICryptoContext
+        {
+            private readonly ISignature _signature;
+            private readonly bool _verifyResult;
+
+            public FakeContext(ISignature signature, bool verifyResult)
+            {
+                _signature = signature;
+                _verifyResult = verifyResult;
+            }
+
+            public int PrivateKeyLength { get; }
+            public int PublicKeyLength { get; }
+            public int SignatureLength { get; }
+            public int SignatureContextMaxLength { get; }
+            public IPrivateKey GeneratePrivateKey() { throw new NotImplementedException(); }
+
+            public IPublicKey GetPublicKeyFromPrivateKey(IPrivateKey privateKey)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'GetPublicKeyFromPrivateKey'.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4600656)